test/plugins/collections.test.js

```diff
@@ -267,6 +302,22 @@ describe('collection plugin test', () => {
             });
         });
 
+        it('returns a collection only with pipelines that exist', () => {
```

Comment:
So if a collection references a pipeline that no longer exists, it will always filter it out on every call to this method. Which means that when a pipeline is removed, all collections that reference it will never update their set of IDs to remove that pipeline.